src/Disks/MetadataStorageFromBackupFile.h

```diff
@@ -0,0 +1,109 @@
+#pragma once
+
+#include <unordered_map>
+#include <vector>
+#include <Disks/ObjectStorages/IMetadataStorage.h>
+#include <Disks/IDisk.h>
+
+
+namespace DB
+{
+
+class MetadataStorageFromBackupFile final : public IMetadataStorage
```

Comment:
A small comment in code about this class here will also be good to have - its purpose and how it works - shortly.